orquesta/utils/context.py

```diff
@@ -40,11 +40,19 @@ def set_current_task(context, task):
     return ctx
 
 
-def set_current_item(context, item):
+def copy_context(context):
     if context and not isinstance(context, dict):
         raise TypeError("The context is not type of dict.")
 
     ctx = json_util.deepcopy(context) if context else dict()
+    return ctx
+
+
+def set_current_item(context, item):
+    if context and not isinstance(context, dict):
+        raise TypeError("The context is not type of dict.")
+
+    ctx = {**context}
```

Comment:
This is a shallow instead of deep copy , correct?

Reply:
I would actually leave this function as is and just do the operation in render.

Reply:
Alright, I'll move it over. And yes it's a shallow copy, but intentional. The rest of the sub-objects that are copied in a deep copy don't actually change between iteration from what I saw in testing, but the top level is what needs to change which a shallow copy still facilitates.

Reply:
@guzzijones thoughts on undoing the changes to `set_current_item`, that function is not used anywhere else in orquesta. `set_current_item` would not work at the current location that `copy_context` is being used, so we could either remove the `set_current_item` function and move the logic into the `render` function, or leave `set_current_item` in the current form

Reply:
@guzzijones thoughts on undoing the changes to `set_current_item`, that function is not used anywhere else in orquesta. `set_current_item` would not work at the current location that `copy_context` is being used, so we could either remove the `set_current_item` function and move the logic into the `render` function, or leave `set_current_item` in the current form